main.go

```diff
@@ -22,34 +22,43 @@ type User struct {
 	Balance uint   `json:"balance"`
 }
 
-// Server represents server with pointer to a DB.
+const (
+	userEnvVar   = "DB_USER"
+	passEnvVar   = "DB_PASS"
+	dbNameEnvVar = "DB_NAME"
+)
+
+// Server represents а db server in social tournaments service.
 type Server struct {
 	DB *sql.DB
 }
 
-func dbConn() (db *sql.DB, err error) {
-	dbDriver := "mysql"
-	dbUser := "root"
-	dbPass := "Sql_DB_tournament1"
-	dbName := "tournament_db"
-
-	db, err = sql.Open(dbDriver, fmt.Sprintf("%s:%s@/%s", dbUser, dbPass, dbName))
+// NewServer constructs a Server, according to existing env variables.
+func NewServer() (*Server, error) {
+	dbUser := os.Getenv(userEnvVar)
+	if dbUser == "" {
+		return nil, fmt.Errorf(`no "%s" env variable`, userEnvVar)
+	}
+	dbPass := os.Getenv(passEnvVar)
+	dbName := os.Getenv(dbNameEnvVar)
+	if dbName == "" {
+		return nil, fmt.Errorf(`no "%s" env variable`, dbNameEnvVar)
+	}
+	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@/%s", dbUser, dbPass, dbName))
 	if err != nil {
 		return nil, fmt.Errorf("can't open db: %s", err)
 	}
-	return db, nil
+	return &Server{db}, nil
```

Comment:
When initializing structs, use filed names to prevent from possible errors in future and make code more clear